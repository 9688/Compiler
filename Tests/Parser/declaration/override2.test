enum TT {A, B, C};
typedef TT TENUM;

int main(){
	struct TENUM
	{} X;
	TT Y;
}