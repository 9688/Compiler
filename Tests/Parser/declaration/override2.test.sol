<<<table #1>>>
<tags>
<types>
int
char
double
void
typedef <TINT> <int
typedef <TT> <int
<vars>
var main type function (){
                 <<<table #3>>>
                 <tags>
                 TINT record
                    <<<table #4>>>
                    <tags>
                    <types>
                    <vars>
                    var x type int
                  endrecord
                 <types>
                 <vars>
                 var X type TINT record
                               <<<table #4>>>
                               <tags>
                               <types>
                               <vars>
                               var x type int
                             endrecord
                 var Y type int
                 ÀÄÄ{empty block}

} returned int

