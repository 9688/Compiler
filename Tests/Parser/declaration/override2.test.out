<--------TABLES-------->
<---TABLE--->
VARS::

main   FUNC () { BLOCK } RETURNED int   


TYPES::
char char

double double

int int

void void

TT ENUM TT {
A
B
C
}

TENUM ENUM TT {
A
B
C
}

<---TABLE--->
VARS::

X   STRUCT TENUM{
vars::

types::
}   


Y   ENUM TT {
A
B
C
}   


TYPES::
TENUM STRUCT TENUM{
vars::

types::
}

